ddp-workspace/projects/ddp-pancan/src/app/components/anchors-page/anchors-page.component.ts

```diff
@@ -0,0 +1,30 @@
+import { ChangeDetectionStrategy, Component, Inject, Input } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { WindowRef } from 'ddp-sdk';
+import { AppRoutes } from '../app-routes';
+
+@Component({
+    selector: 'app-anchors-page',
+    templateUrl: './anchors-page.component.html',
+    styleUrls: ['./anchors-page.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class AnchorsPageComponent {
+    @Input() source: string;
+    @Input() routerLink: string;
+    anchorLabel = 'section';
+    readonly AppRoutes = AppRoutes;
+
+    constructor(
+        private window: WindowRef,
+        @Inject(DOCUMENT) private document: Document
+    ) {}
+
+    scrollToAnchor(anchor: string): void {
+        const element = this.document.getElementById(anchor);
+        if (element) {
+            const y = element.getBoundingClientRect().top;
+            this.window.nativeWindow.scrollTo({top: y, behavior: 'smooth'});
```

Comment:
wondering what's the reason we need to programmatically scroll? it should be scrolled by the native browser behavior

Reply:
Was thinking about scroll to a bit higher point then the element itself as to see full content.
Anyway replaced.